src/test/FuzzerImpl.cpp

```diff
@@ -838,70 +906,325 @@ applyFuzzOperations(LedgerTxn& ltx, PublicKey const& sourceAccount,
+        : SponsoredEntryParameters(true, sponsorKey)
+    {
+    }
+
+    bool const mSponsored;
+    int const mSponsorKey; // meaningful only if mSponsored is true
+
+  private:
+    constexpr SponsoredEntryParameters(bool sponsored, int sponsorKey)
+        : mSponsored(sponsored), mSponsorKey(sponsorKey)
+    {
+    }
+};
+
+struct AccountParameters : public SponsoredEntryParameters
+{
+    constexpr AccountParameters(int shortKey,
+                                int64_t nativeAssetAvailableForTestActivity,
+                                uint32_t optionFlags)
+        : SponsoredEntryParameters()
+        , mShortKey(shortKey)
+        , mNativeAssetAvailableForTestActivity(
+              nativeAssetAvailableForTestActivity)
+        , mOptionFlags(optionFlags)
+    {
+    }
+
+    constexpr AccountParameters(int shortKey,
+                                int64_t nativeAssetAvailableForTestActivity,
+                                uint32_t optionFlags, int sponsorKey)
+        : SponsoredEntryParameters(sponsorKey)
+        , mShortKey(shortKey)
+        , mNativeAssetAvailableForTestActivity(
+              nativeAssetAvailableForTestActivity)
+        , mOptionFlags(optionFlags)
+    {
+    }
+
+    int const mShortKey;
+    int64_t const mNativeAssetAvailableForTestActivity;
+    uint32_t const mOptionFlags;
+};
 
 std::array<
     AccountParameters,
     FuzzUtils::NUMBER_OF_PREGENERATED_ACCOUNTS> constexpr accountParameters{
-    {{DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, true},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, true},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, true},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, true},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false}}};
-
-// Unlike Asset, this can be a constexpr.
-struct AssetID
+    {{0, 256, 0},
+     {1, 256, 0},
+     {2, 256, 0, 1}, // sponsored by account 1
+     {3, 256, AUTH_REQUIRED_FLAG},
+     {4, 256, 0},
+     {5, 256, 0},
+     {6, 256, 0},
+     {7, 256, 0},
+     {8, 256, 0},
+     {9, 256, 0},
+     {10, 256, 0},
+     {11, 256, 0},
+     {12, 256, 0},
+     {13, 256, 0},
+     {14, 256, 0},
+     {15, 256, 0}}};
+
+struct TrustLineParameters : public SponsoredEntryParameters
 {
-    constexpr AssetID() : mIsNative(true), mId(0)
+    constexpr TrustLineParameters(int trustor, AssetID assetID,
```

Comment:
Nit: `assetID` can be `const&` here and in the methods below.

Reply:
Changed.